include/devsettings_widgets.hpp

```diff
@@ -0,0 +1,360 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2009-2010 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#ifndef _DEVSETTINGS_WIDGETS_H
+#define _DEVSETTINGS_WIDGETS_H
+
+#include <QDialog>
+#include <QGroupBox>
+#include <QWidget>
+
+class QBoxLayout;
+class QGridLayout;
+
+class QLabel;
+class QLineEdit;
+class QCheckBox;
+class QComboBox;
+class QPushButton;
+class QRadioButton;
+class QTreeWidget;
+class QTreeWidgetItem;
+class QButtonGroup;
+class QSlider;
+
+class TextSlider;
+class device;
+
+class devctlAutostrategy;
+class TattooWidget;
+
+class devctlCommon : public QGroupBox {
+  Q_OBJECT
+
+public:
+  devctlCommon(device *idev, QWidget *p = NULL);
+  ~devctlCommon(){};
+
+private slots:
+  void set_prec(bool);
+  void set_hcdr(bool);
+  void set_ss(bool);
+  void set_spdrd(bool);
+  void set_bitset(bool);
+  void set_bitset_dl(bool);
+  void set_simulplus(bool);
+
+private:
+  device *dev;
+
+  QGridLayout *layout;
+  QCheckBox *ck_prec;
+  QLabel *pl_prec;
+  QLabel *l_prec;
+
+  QCheckBox *ck_hcdr, *ck_ss, *ck_spdrd, *ck_bitset, *ck_bitset_dl,
+      *ck_simulplus;
+
+  devctlAutostrategy *as;
+};
+
+// GigaRec control
+
+class devctlGigarec : public QGroupBox {
+  Q_OBJECT
+public:
+  devctlGigarec(device *idev, QWidget *p = NULL);
```

Comment:
Codacy found an issue: [Class 'devctlGigarec' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)